tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
                 auto bot_tot = fmod(bot->GetTimeTrailing_ns() - bot->GetTimeLeading_ns() + c_range_ns, c_range_ns);
                 // register multi hits
                 multihits[iPlane - 1][iBar - 1] += 1;
+				
+				Double_t Ctop_tot = 0./0.;
+				Double_t Cbot_tot = 0./0.;
 
-                if (fParameter == 1)
+				Bool_t ToTexeption = false;
+                
+				if (fParameter == 1)
                 {
                     // calculate tdiff
                     auto tdiff = (bot_ns) - (top_ns);
-
+/*
+				if(iPlane == 2 && iBar == 7){
+					if(sqrt(top_tot*bot_tot) > fToTMin && sqrt(top_tot*bot_tot) < 500.) ToTexeption = true;
+				}
+				if(iPlane == 2 && iBar == 1){
+					if(sqrt(top_tot*bot_tot)> fToTMin && sqrt(top_tot*bot_tot) < 350.) ToTexeption = true;
+				}
+				if(iPlane == 2 && iBar == 9){
+					if(sqrt(top_tot*bot_tot)>fToTMin && sqrt(top_tot*bot_tot) < 500.) ToTexeption = true;
+				}
+*/
                     // fill control histograms
                     fhLogTot1vsLogTot2[iPlane - 1][iBar - 1]->Fill(TMath::Log(top_tot), TMath::Log(bot_tot));
-                    fh_tofd_TotPm[iPlane - 1]->Fill(iBar, top_tot);
-                    fh_tofd_TotPm[iPlane - 1]->Fill(-iBar - 1, bot_tot);
+                if((sqrt(top_tot*bot_tot)>fToTMin && sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption){    
+					fh_tofd_TotPm[iPlane - 1][0]->Fill(iBar, top_tot);
+                    fh_tofd_TotPm[iPlane - 1][0]->Fill(-iBar - 1, bot_tot);
+					fh_tofd_TotPm1[iPlane - 1]->Fill(iBar, top_tot);
+					fh_tofd_TotPm1[iPlane - 1]->Fill(-iBar - 1, bot_tot);
+				}
+				if(/*top_tot>10. && */top_tot<90.){
+					fh_tofd_TotPm2[iPlane - 1]->Fill(iBar, top_tot);
+					cout<<"Fill Top ToT "<<top_tot<<endl;
+				}
+				if(/*bot_tot>10. && */bot_tot<90.){
+					fh_tofd_TotPm2[iPlane - 1]->Fill(-iBar - 1, bot_tot);
+					cout<<"Fill Bot ToT "<<bot_tot<<endl;
+				}
 
                     // Time differences of one paddle; offset  histo
-                    fhTdiff[iPlane - 1]->Fill(iBar, tdiff);
-
+                    if((sqrt(top_tot*bot_tot)>fToTMin &&sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption){
+		    fhTdiff[iPlane - 1]->Fill(iBar, tdiff);
+		    }
                     auto posToT = TMath::Log(top_tot / bot_tot);
-                    fhSqrtQvsPosToT[iPlane - 1][iBar - 1]->Fill(posToT, sqrt(top_tot * bot_tot));
-
+                   if((sqrt(top_tot*bot_tot)>fToTMin &&sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption){
```

Comment:
There are 3 identical if statements (if((sqrt(top_tot*bot_tot)>fToTMin &&sqrt(top_tot*bot_tot)<fToTMax) || ToTexeption)). Why?

ToTexeption (spelt incorrectly) is applied, but it's not clear why. When would the ToTexeption condition be added and when not?

If it's always false, why not just remove it?